packages/replayio/src/utils/graphql/queryGraphQL.ts

```diff
@@ -0,0 +1,31 @@
+import fetch from "node-fetch";
+import { graphQLServer } from "../../config";
+import { getUserAgent } from "../getUserAgent";
+import { debug } from "./debug";
+
+export async function queryGraphQL(name: string, query: string, variables = {}, apiKey?: string) {
+  const options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "User-Agent": getUserAgent(),
+    } as Record<string, string>,
+    body: JSON.stringify({
+      query,
+      name,
+      variables,
+    }),
+  };
+
+  if (apiKey) {
+    options.headers.Authorization = `Bearer ${apiKey.trim()}`;
+  }
+
+  debug("Querying %s over %s graphql endpoint", name, graphQLServer);
+  const result = await fetch(`${graphQLServer}/v1/graphql`, options);
+
+  const json: any = await result.json();
+  debug("GraphQL Response: %O", json);
+
+  return json;
```

Comment:
what do you think about always throwing `GraphQLError` here when `json.errors` are present?

Reply:
I _think_ I lean towards letting the caller handle `response.error` 🤔  because that seems to be more idiomatic with other GraphQL uses I've seen, though admittedly I'm not super experienced with GraphQL so I could be convinced otherwise!